netbox/netbox/api/authentication.py

```diff
@@ -123,6 +124,32 @@ def has_object_permission(self, request, view, obj):
 
         return super().has_object_permission(request, view, obj)
 
+    # Helper function to obtain the desired action from the first permission in the permission map
+    def get_action(self, method):
+        # Get the first permission
+        permission = self.perms_map.get(method)[0] if len(self.perms_map.get(method)) > 0 else None
+        if permission:
+            # Remove app and model label
+            app_label, action, model_name = resolve_permission(permission)
+            return action
+        return None
+
+
+class RequireViewOnlyPermissions(TokenPermissions):
+    """
+    Overrides permission map to return only view permissions as required
+    """
+    # Override the stock perm_map to enforce view permissions
+    perms_map = {
```

Comment:
This approach isn't sufficient as it doesn't permit the use of a write-disabled API token. We need to override `has_permission()` and `has_object_permission()` on the permission class.